lift/Readme.md

```diff
@@ -6,271 +6,360 @@
+migrations.
 
-- Prisma can make sure, that the local migrations are in sync with the databases migrations
-- Prisma can ensure, that migrations don't get run twice
-- Prisma can give an overview in a GUI interface about the currently running migration and migrations that already have been running
-- Prisma can rollback migrations when there has been an error
+```
+┌──────────────────┐       ┌──────────────────┐      ┌──────────────┐
+│                  │       │                  │      │   Postgres   │
+│   Lift Client    │       │   Lift Server    │     ┌┴────────────┬─┘
+│                  │       │                  │     │    MySQL    │
+└──────────────────┘       └──────────────────┘     └────────┬────┘
+          │          save            │                     │ │
+          ├────────────┐             │        infer        │ │
+          │            └────────────▶│──────────┬┐         │ │
+          │                          │          └┼────────▶│ │
+          │                          │           └─────────┼▶│
+          │                          │      migration      │ │
+          │        migration         │          ┌──────────┤ │
+          │            ┌─────────────│◀─────────┴──────────┼─┤
+          │◀───────────┘             │                     │ │
+          │                          │                     │ ▼
+          ▼                          ▼                     ▼
+```
 
-## Detailed design
+### Lift Client
 
-As mentioned in the [basic example](#basic-example), a simple `prisma migrate` is needed to migrate the database and create a new migration.
+The Lift Client is built into the Prisma 2 CLI but is also accessible programmatically in the Prisma SDK. The Lift Client has 3 methods:
 
-These are the commands:
+#### Save
 
-```bash
-prisma migrate                                Creates a migration, clears the draft and migrates the database
+Save method takes the difference of your schema file with the current state of your datasources and creates a migration folder. This migration folder contains
+the steps to sync your datasources with your schema file. The template of the migration folder is `${current_timestamp}-${migration-name}/`
 
-       migrate plan                           Creates a migration without applying it
-       migrate apply                          Applies all unapplied migrations
-       migrate draft                          Pushes the new changes into a draft without creating a migration
-       migrate rollback                       Rolls back migration(s)
+```
+migrations/
+  └─ 20190920142118-initial/
+    └─ steps.json
+    └─ schema.prisma
+    └─ README.md
 ```
 
-Let's discuss a few questions.
+A migration contains 3 files:
 
-### Naming of migration folders
+- **steps.json:** contains a JSON list of steps to run against the database
```

Comment:
Both the up and down versions? Or are those the sames, and the different result is just calculated when executing up or down?

Reply:
They are the same. We calculate up and down on the fly.